src/Tools/ExternalAccess/LiveShare/RemoteLanguageServiceWorkspaceHost.cs

```diff
@@ -111,6 +117,16 @@ public async Task EnsureProjectsLoadedAsync(CancellationToken cancellationToken)
             }
         }
 
+        /// <summary>
+        /// Initialize the workspace.  This must be done on the UI thread because it
+        /// could potentially attempt to create services that require the UI thread. <see cref="LanguageSettingsPersister"/>
+        /// </summary>
+        private async Task InitWorkspaceAsync(CancellationToken cancellationToken)
+        {
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+            _remoteLanguageServiceWorkspace.Init();
```

Comment:
another approach could be what @jasonmalinowski did for IExperiementService.

put 
// Ensure the options persisters are loaded since we have to fetch options from the shell
_componentModel.GetExtensions<IOptionPersister>();
(http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/RoslynPackage.cs,75) in liveshare explicitly so that we make sure option is initialized before liveshare use it.

...

all these used to be in RoslynPackage, but with async project load, RoslynPackage is no longer loaded until a file open, so, these initialization now need to be moved to somewhere else. but unlike jason's case where solution creation is the starting point, for us (liveshare), live share service creation is the starting point. solution that belong to livesharepworkspace is not added yet.